src/llama-sampling.cpp

```diff
@@ -356,6 +356,19 @@ struct llama_sampler * llama_sampler_chain_get(const struct llama_sampler * chai
     return p->samplers[i];
 }
 
+struct llama_sampler * llama_sampler_chain_remove(struct llama_sampler * chain, int32_t i) {
+    auto * p = (llama_sampler_chain *) chain->ctx;
+
+    if (i < 0 || i >= (int32_t) p->samplers.size()) {
```

Comment:
Since it is already established that `i >= 0` in the first comparison, casting `i` to `size_t` instead removes the possibility of an overflow.
```suggestion
    if (i < 0 || (size_t) i >= p->samplers.size()) {
```

Reply:
Fixed